### DISCUSSION TITLE
Can a single extension crash stop the whole theia application

### DISCUSSION THREAD

**Discussion by jeraldpraveen** [eclipse-theia/theia#13389] [Q&A] [answered]
Whenever an extension crashes in theia, I am getting the following error message and theia application has to be restarted.
![image](https://github.com/eclipse-theia/theia/assets/72866500/868ec7fe-af4c-44d8-a806-7c54697e773c)

Is there any event to capture extension crash for reporting to app insights.

**Comment by msujew** [comment, marked as answer]
Hey @jeraldpraveen,

> Can a single extension crash stop the whole theia application

there's nothing Theia can really do if an extension decides to crash the whole plugin host. However, extensions usually shouldn't do that. Do you have a specific extension that regularly crashes? Note that the application doesn't stop working, only the extension support.

> Is there any event to capture extension crash for reporting to app insights.

The IPC error is logged, see [here](https://github.com/eclipse-theia/theia/blob/63d66679f5a681a83affb7361333134d5d55f8cd/packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts#L218), but there's no event triggered. Feel free to contribute something to improve on that.
